src/main/java/org/folio/search/utils/CallNumberUtils.java

```diff
@@ -202,8 +202,11 @@ public static List<CallNumberBrowseItem> excludeIrrelevantResultItems(BrowseCont
         if (instance != null) {
           return instance.getItems()
             .stream()
-            .anyMatch(i -> r.getFullCallNumber() == null
-              || getFullCallNumber(i).equals(r.getFullCallNumber()));
+            .anyMatch(i -> {
+              String fullCallNumber = getFullCallNumber(i);
+              return r.getFullCallNumber() == null
+                || fullCallNumber != null && fullCallNumber.equals(r.getFullCallNumber());
+            });
         }
```

Comment:
looks like, from the point of better readability it makes sense to inline this block of code served as a predicate in ```.filter(...)``` to some method